plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/VarbinaryToVarcharCoercers.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.plugin.hive.HiveStorageFormat;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.VarbinaryType;
+import io.trino.spi.type.VarcharType;
+
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CodingErrorAction;
+import java.util.HexFormat;
+
+import static io.trino.plugin.hive.HiveStorageFormat.ORC;
+import static io.trino.plugin.hive.HiveStorageFormat.PARQUET;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
+import static io.trino.spi.type.Varchars.truncateToLength;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class VarbinaryToVarcharCoercers
+{
+    private VarbinaryToVarcharCoercers() {}
+
+    public static TypeCoercer<VarbinaryType, VarcharType> createVarbinaryToVarcharCoercer(VarcharType toType, HiveStorageFormat storageFormat)
+    {
+        if (storageFormat == ORC) {
+            return new OrcVarbinaryToVarcharCoercer(toType);
+        }
+        if (storageFormat == PARQUET) {
+            return new ParquetVarbinaryToVarcharCoercer(toType);
+        }
+        return new VarbinaryToVarcharCoercer(toType);
+    }
+
+    public static class VarbinaryToVarcharCoercer
```

Comment:
Do the implementations (VarbinaryToVarcharCoercer,  ParquetVarbinaryToVarcharCoercer, OrcVarbinaryToVarcharCoercer) need to be public if we use factory method and don't construct them explicitly by name?